### DISCUSSION TITLE
add method to return both content and metadata?

### DISCUSSION THREAD

**Discussion by unsleepy22** [adbar/trafilatura#758] [Ideas]
We now have two extraction methods in core: `bare_extraction` and `extract` . The former returns internal structure while the latter returns only parsed content.  In many cases we need to retrieve both web page metadata & content and store them separately. It does not feel elegant to parse metadata from content (when we set with_metadata=True), we wouldn't want to parse metadata in a separate call either. So is it possible that we add a method like `extract_with_metadata` , to return `Union[Document, str]` ?

**Comment by unsleepy22** [comment]
@adbar what do you think?

**Reply by adbar** [reply]
Since the `with_metadata` already exists the new function would not be an improvement, just a wrapper around the existing function with `with_metadata=True`, right?

**Reply by unsleepy22** [reply]
yes I think so
